### PR TITLE
FEATURE #3335 single event request and subscription support

### PR DESCRIPTION
**Pull request by tiaki75019**
Hello @encoreshao 

this PR contains support to:
- GET single event -> `/me/events/{id}`
- POST subscription -> `/v1.0/subscriptions` (create webhook for O365)
- PATCH subscription -> `/v1.0/subscriptions/{identifier}` (update webhook expiresDate)

please have a look and let me know if any concerns

~~PS: Drafting it, I am making some tests in local development.~~

### CODE REVIEW

**Comment by tiaki75019** [comment]
can be reviewed

File: lib/office365/rest/event.rb [hunk 1]
```diff
@@ -12,6 +12,12 @@ module Event
       def events(args = {})
         wrap_results(args.merge(kclass: Models::Event, base_uri: "/me/events"))
       end
+
+      def event(identifier)
+        raise ArgumentError, "Identifier must be provided" if identifier.nil? || identifier.empty?
+
+        wrap_results(kclass: Models::Event, base_uri: "/me/events/#{identifier}", get_by_id: true)
```
Comment (encoreshao) [inline lib/office365/rest/event.rb]: @tiaki75019 

let's add an variable for base URI

```
BASE_URI = "/me/events".free

....

[BASE_URI, identifier].join('/')
```

Shall we use identifier to decide the response? I think it's enough

```
wrap_results(kclass: Models::Event, base_uri: [BASE_URI, identifier].join('/'), identifier: identifier)
```

Comment (tiaki75019) [reply lib/office365/rest/event.rb]: you are right

```
wrap_results(kclass: Models::Event, base_uri: [BASE_URI, identifier].join('/'), identifier: identifier)
```

File: lib/office365/rest/request.rb [hunk 2]
```diff
@@ -29,12 +29,26 @@ def get(uri, args: {})
...
+
+      def patch(uri, args)
+        req_url = URI(uri.start_with?("https") ? uri : (Office365::API_HOST + uri))
+        json_header = args.delete(:json_header)
+
+        response = Faraday.new(url: [req_url.scheme, "://", req_url.hostname].join, headers: json_header ? headers : post_headers) do |faraday|
+          faraday.adapter Faraday.default_adapter
+          faraday.response :json
+          faraday.response :logger, ::Logger.new($stdout), bodies: true if dev_developement?
+        end.patch(req_url.request_uri, json_header ? args.to_json : args.ms_hash_to_query)
```
Comment (encoreshao) [inline lib/office365/rest/request.rb]: `post vs patch` - there seems to be only one difference, let make some refactor here.


```
api_changed_operations(__method__)

def api_changed_operations(method_name)
  faraday = Faraday.new(url: [req_url.scheme, "://", req_url.hostname].join, headers: json_header ? headers : post_headers) do |faraday|
    .....
  end

  faraday.send(method_name, req_url.request_uri, json_header ? args.to_json : args.ms_hash_to_query)
end
```

Comment (tiaki75019) [reply lib/office365/rest/request.rb]: thanks for the tips

done 


called the method `faraday_action`

File: lib/office365/rest/event.rb (hunk 1 above)
Comment (tiaki75019) [reply lib/office365/rest/event.rb]: done

**Pull request merged**